### PR TITLE
fix(dashboard): make to show the correct owned objects

### PR DESCRIPTION
**Pull request by prosdev0107**
### SUMMARY
Making myself owner of a dashboard that was NOT created by me does not show in "Mine" tab on homepage

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
BEFORE:

https://user-images.githubusercontent.com/47900232/158665539-27d39b54-a5a9-4207-83b7-0cf06a85ed34.mov

AFTER:

https://user-images.githubusercontent.com/47900232/158665158-be40dd12-9c51-4892-ad83-442266d15cc7.mov

### TESTING INSTRUCTIONS
**How to reproduce bugs**

1. Click the "Dashboards" tab
2. Hover over the actions panel
3. Click the "Edit" button
4. Make yourself the owner of the dashboard
5. Click the "Save" button
6. Go back to the "Home" page
7. Select the "Mine" section in "Dashboard" panel
8. Pay attention to the dashboards

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/utils.tsx
Comment (diegomedina248) [inline superset-frontend/src/views/CRUD/utils.tsx:170]: this kind of defeats the purpose to me, and makes this function harder to reuse.
I would add a `col` and `opr` as third and fourth param, assigning the 'owners' and 'rel_m_m' as defaults.
Then, when calling for `saved_query`, we can override those, and reuse it for other calls if needed.

**Review by diegomedina248 — CHANGES_REQUESTED** [review]

**Comment by prosdev0107** [comment]
@diegomedina248 

Can you review this PR and approve if it is good?

**Review by diegomedina248 — APPROVED** [review]

**Review by rusackas — APPROVED** [review]
LGTM

**Comment by villebro** [comment]
@prosdev0107 @diegomedina248 After this PR I'm getting this error toast on the Welcome page: 
image

**Comment by diegomedina248** [comment]
@prosdev0107 please review, I am able to reproduce the error as well.
It happens on initial load, if you then switch filters it works.

**Comment by prosdev0107** [comment]
@diegomedina248 
I checked the this part code and I found that someone(I could see it from git history) made changes in this part after merging this PR.
I think that someone made the changes when rebase or merge master into his branch.

**Pull request merged**
